modules/roles_profiles/manifests/profiles/cltbld_user.pp

```diff
@@ -14,12 +14,19 @@
             # Create the cltbld user
             users::single_user { 'cltbld':
                 # Bug 1122875 - cltld needs to be in this group for debug tests
-                groups     => [ '_developer' ],
                 password   => $password,
                 salt       => $salt,
                 iterations => $iterations,
             }
 
+            # Monkey patching directoryservice.rb in order to create users also breaks group merging
+            # So we directly add the user to the group(s)
+            exec { 'cltbld_developer_group':
+                command => '/usr/bin/dscl . -append /Groups/_developer GroupMembership cltbld',
+                unless  => '/usr/bin/groups cltbld | /usr/bin/grep -q -w _developer',
```

Comment:
you'd mentioned needing to do this. Is this maybe more maintainable than a patch anyway?

Reply:
The patch is still needed to create the users through puppet and I'm hoping puppetlabs will have the bug fixed soon.  We might be able to work some magic by signing the launchd provisioning script with our developer installation key if looks like the bug won't be fixed in a timely manner.  But for right now, this just works.